docs/05-terraform.md

```diff
@@ -15,4 +15,192 @@ This Terraform script can be used to deploy an entire environment from scratch.
+4. (Optional) If you have changed the database name from the default `finspace-database` to any other names, please also edit the `env.q` file, changing the database name to the new variable that you have set in line 19.
+5. Run `aws configure` in the terminal to set up your access key and secret key from your AWS account. This is needed to connect to your account and use the Terraform deployment. Check our resource link for more instructions on how to find your access key and secret key.
+6. From your Terraform working directory which is `TorQ-Amazon-FinSpace-Starter-Pack/terraform-deployment/deployments`, run `terraform init`.
+7. If initialized without error, run `terraform plan`. This will show all resources set to be created or destroyed by Terraform.
+8. Run `terraform apply` to execute this plan. The initial deployment can take approximately 45 minutes, and connection losses can cause errors with deployment, so it's a good idea to run this in `nohup`. (Using `nohup` might lead to a higher cost of operating the codes if you are using Terraform from a cloud environment.)
+
+You can now skip ahead to our [Managing Your Infrastructure section](#managing-your-infrastructure)
+
+## How to Use - Deploying With Terraform For Users With Existing Infrastructure
+
+For users with existing infrastructure in their AWS account who would like to reuse the same resources for their TorQ in Finspace bundle, you can use import blocks in Terraform. This functionality allows you to import existing infrastructure resources into Terraform, bringing them under Terraform's management. The import block records that Terraform imported the resource and did not create it. After importing, you can optionally remove import blocks from your configuration or leave them as a record of the resource's origin.
+
+Once imported, Terraform tracks the resource in your state file. You can then manage the imported resource like any other, updating its attributes and destroying it as part of a standard resource lifecycle.
+
+Move into the `deployments` directory, and you'll see an `imports.tf` file (currently empty). This `imports.tf` file is automatically run before Terraform applies any changes to the structure, importing existing structures from your AWS to the deployment system.
+
+### Terraform Import Block Syntax
+
+```
+ import {
+  to = aws_instance.example
+  id = "i-abcd1234"
+}
+
+resource "aws_instance" "example" {
+  name = "hashi"
+  # (other resource arguments...)
+}
+```
+The above `import` block defines an import of the AWS instance with the ID "i-abcd1234" into the `aws_instance.example` resource in the root module.
+
+The import block has the following arguments:
+
+`to` - The instance address this resource will have in your state file.
+`id` - A string with the import ID of the resource.
```

Comment:
This can be misleading - it should be any attribute that informs terraform about the exact resource, as not all resources are assigned an id

For example, if importing an existing cloudwatch log group 
```
import {
   to = aws_cloudwatch_log_group.myclustername
   name = "aws/vendedlogs/finspace/myclustername"
}
```